src/clj/rems/application/model.clj

```diff
@@ -663,7 +665,10 @@
     (update application :application/attachments #(filterv visible? %))))
 
 (defn see-application? [application user-id]
-  (not= #{:everyone-else} (permissions/user-roles application user-id)))
+  (let [permissions (permissions/user-roles application user-id)]
+    (if (and (= :application.state/draft (:application/state application)) (contains? permissions :reporter))
+      false
+      (not= #{:everyone-else} permissions))))
```

Comment:
I feel like this logic could be a bit more clear. Now it disallows reporters from seeing drafts. I'd rather only allow applicants (and members?) to see drafts. So something like

```
(if draft
  (user-is-applicant-or-member)
  (not= #{:everyone-else permissions))

Reply:
I'd still like to see this clarified